File: solutions/typescript/resistor-color-trio/1/resistor-color-trio.ts
export function decodedResistorValue(colors: string[]): string {
  const significantDigits = decodedValue(colors.slice(0,2));
  const factor = Math.pow(10,colorCode(colors[2]));
  var prefix = '';


  var value = significantDigits * factor;
  if (value >= 1_000_000_000) {
    value /= 1_000_000_000;
    prefix = 'giga';
  } else if (value >= 1_000_000) {
    value /= 1_000_000;
    prefix = 'mega';
  } else if (value >= 1_000) {
    value /= 1_000;
    prefix = 'kilo';
  }

  return `${value} ${prefix}ohms`;
}

export function decodedValue(colors: string[]): number {
  return (colorCode(colors[0]) * 10) + colorCode(colors[1]) ;
}

export const colorCode = (color: string): number => {
  return COLORS.indexOf(color);
}

export const COLORS = [
  'black',
  'brown',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'violet',
  'grey',
  'white',
];

File: solutions/python/resistor-color-trio/1/resistor_color_trio.py
COLORS = [
    "black",
    "brown",
    "red",
    "orange",
    "yellow",
    "green",
    "blue",
    "violet",
    "grey",
    "white"
]

MAGNITUDES = [
    "",
    "kilo",
    "mega",
    "giga"
]

def label(colors):
    digits = 0

    for i in range(0, 2):
        digits += COLORS.index(colors[i]) * pow(10, 1-i)

    digits *= pow(10, COLORS.index(colors[2]))

    magnitude = 0
    while digits >= 1000:
        digits /= 1000
        magnitude += 1

    return str(int(digits)) + " " + MAGNITUDES[magnitude] + "ohms"
